Documentation/ApiOverview/Environment/ApplicationContext.rst

```diff
@@ -0,0 +1,383 @@
+
+    *   Errors and warnings are logged.
+    *   Obsolete (deprecated) function calls are logged in an additional
+        log file.
+    *   The error appears in the frontend with a note on where and in which
+        file it occurred.
+    *   In the backend, the corresponding table field is also displayed in
+        square brackets after each label when editing data records.
+    *   The :guilabel:`Clear All Cache` button at the top right also clears
+        the :ref:`Dependency injection <Dependency-Injection>` cache.
+    *   The menu items in the backend for the install tool no longer require an
+        additional password entry.
+    *   Admins without system maintainer authorisation can also see the menu
+        items for the install tool.
+
+*   Testing
+
+    *   In this special mode, caching
+        for :ref:`Class Loading <composer-class-loader>` is switched off or is
+        only valid for one request.
+
+..  _default-application-context:
+
+Default ApplicationContext
+==========================
+
+In the :php:`\TYPO3\CMS\Core\Core\SystemEnvironmentBuilder`, TYPO3 reads
+different environment variables to determine the desired `ApplicationContext`.
+The following order applies from high priority to no priority:
+
+*   Environment variable: :php:`TYPO3_CONTEXT`
+*   Environment variable: :php:`REDIRECT_TYPO3_CONTEXT`
+*   Environment variable: :php:`HTTP_TYPO3_CONTEXT`
+*   `ApplicationContext` is set to `Production`
+
+..  _set-application-context:
+
+Set the ApplicationContext
+==========================
+
+The `ApplicationContext` is read very early in TYPO3. Even before the actual
+TYPO3 bootstrap process. This means that even if you set the
+`ApplicationContext` in `additional.php` (formerly
+`AdditionalConfiguration.php`), it is already too late.
+
+It is best to set the `ApplicationContext` as an environment variable in the
+server configuration. Alternatively, you need a solution to set the environment
+variable at PHP level before the actual TYPO3 call.
+
+..  _set-application-context-apache:
+
+Apache
+------
+
+In the vhost configuration as well as in the :file:`.htaccess`, the
+`ApplicationContext` can be set as follows:
+
+..  code-block:: bash
+    :caption: `[web root]/.htaccess`
+
+    SetEnv TYPO3_CONTEXT Development
+
+..  attention::
+    Some hosts may prevent the setting of environment variables using
+    `.htaccess`. In this case, you will have to find another solution. Have a
+    look at the following examples below.
+
+..  _set-application-context-nginx:
+
+Nginx
+-----
+
+..  code-block::
+    :caption: /etc/nginx/nginx.conf
+
+    fastcgi_param TYPO3_CONTEXT Development;
+
+..  _set-application-context-env:
+
+.env (Composer only)
+--------------------
+
+It is possible to import `.env` files into the root directory of your project.
+All contained values are then made available as environment variables. The
+basis for this is the Symfony `.env loader` :composer:`symfony/dotenv`. However,
```

Comment:
There is also https://packagist.org/packages/vlucas/phpdotenv - when you already mention two packages.

Reply:
Don't get the point